src/mds/MDSMap.cc

```diff
@@ -687,6 +678,11 @@ MDSMap::availability_t MDSMap::is_cluster_available() const
     return AVAILABLE;
   } else {
     // Nothing indicating we were stuck, but nobody active (yet)
-    return TRANSIENT_UNAVAILABLE;
+    //return TRANSIENT_UNAVAILABLE;
+
+    // Because we don't have standbys in the MDSMap any more, we can't
+    // reliably indicate transient vs. stuck, so always say stuck so
+    // that the client doesn't block.
+    return STUCK_UNAVAILABLE;
```

Comment:
I think we'd do it externally: at some point (probably at same time as extending caps) we would add a monclient->mon call that enabled it to resolve filesystem names into FSCIDs, and we could provide an advisory flag from the mon in that response that indicated whether the filesystem was available.